### PR TITLE
added MCMC for makemoons

### PR DESCRIPTION
**Pull request by AnandShegde**
Created BNN with blackjax and flax and produced the following results.
image

### CODE REVIEW

File: notebooks/classification/mcmc_bnn_makemoons.ipynb [hunk 1]
```diff
@@ -0,0 +1,462 @@
+{
```
Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    from flax.core import unfreeze
</pre><p>There seem to be two unfreeze imports.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #15.    import optax  # Optimizers
</pre><p>please remove optax and similar unused imports</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    from jax.flatten_util import ravel_pytree
</pre><p>The import shouldn't be here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    def bnn_log_joint(params, X, y, model):
</pre><p>It would be useful to provide a docstring for what params is</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.        logits = model.apply(params, X).ravel()
</pre><p>It would be good to add small comments. like, model.apply return N, 1, ... and we flatten it to ..</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        flatten_params, _ = ravel_pytree(params)
</pre><p>similar to above..would be good to document what flattened_params would look like</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <p>Colorbar missing</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    one_state = tree_map(lambda x:x[i],states.position)
</pre><p>Unclear what is meant by <code>one_state</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    logits = model.apply(one_state, X)
</pre><p>code here and in the below cell is largely same. Can we use some simple utilities and avoid code duplication?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #22.    ax[1].legend(*hs.legend_elements(), fontsize=20)
</pre><p>Titles missing in the plots.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (nipunbatra) [inline notebooks/classification/mcmc_bnn_makemoons.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    plot_binary_class(X_scatters=X,y_scatters=y_hat,XX1_grid=XX1,XX2_grid = XX2,grid_preds_mean=Z,grid_preds_sigma= sigma_Z,titles=("Mean of Predictions of 100  MCMC states","Variance of Predictions of 100  MCMC states"))
</pre><p>Perhaps there are other colormaps that we can use for variance?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nipunbatra/bnn-zoo/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Comment by AnandShegde** [comment]
Sir, I have adopted the suggestions.
